pkg/k8sclient/client.go

```diff
@@ -157,10 +161,18 @@ func inClusterConfig() (*rest.Config, error) {
 
 func outOfClusterConfig() (*rest.Config, error) {
 	kubeconfig := os.Getenv(k8sutil.KubeConfigEnvVar)
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	kubecontext := os.Getenv(k8sutil.KubeContextEnvVar)
+	config, err := buildConfigWithContext(kubeconfig, kubecontext)
 	return config, err
 }
 
+// see clientcmd.BuildConfigFromFlags()
+func buildConfigWithContext(kubeconfigPath string, context string) (*rest.Config, error) {
```

Comment:
FY: this is the most important change that allows me to implement what I need, the rest is just supporting it

Reply:
from the looks of it, decorating this function https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/config/config.go#L59 should give me the same result (after controller-runtime refactoring is merged)

Reply:
I agree with @deads2k 's comment. Any chance you could use https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/genericclioptions/config_flags.go here?

Reply:
sorry, the first link gives me 404

Reply:
> sorry, the first link gives me 404

Fixed: https://github.com/kubernetes/cli-runtime/blob/master/pkg/genericclioptions/config_flags.go